Tensile/KernelWriterAssembly.py

```diff
@@ -3035,25 +3031,27 @@ def globalReadDo(self, kernel, guardK, tP):
       tmpSgpr = maxAddrSgpr + 2 # 7 sgprs available
       #dumpVgpr = self.vgprPool.checkOut(1)
 
-      # maxAddrSgpr = size[n] * stride[n-1]
-      kStr += self.comment1("max read address = size[n] * stride[n-1]")
-      dim = len(tP["ia"])-1 # dim
-      strideIdx = dim-1 # largest stride
-      sizeIdx = tP["ia"][dim]
-
-      sizeIdxIsSum = sizeIdx in kernel["ProblemType"]["IndicesSummation"]
-      if sizeIdxIsSum:
-        sizeIdx -= kernel["ProblemType"]["NumIndicesC"]
-
-      kStr += inst("s_mul_i32", \
-          sgpr(maxAddrSgpr+0), \
-          sgpr("Sizes%s+%u"%("Sum" if sizeIdxIsSum else "Free", sizeIdx)),  \
-          sgpr("Strides%s+%u"%(tP["tensorChar"],strideIdx)), \
-          "mul d%u lower"%dim)
-
-      if kernel["BufferLoad"]:
-        offsetVgpr = self.vgprPool.checkOut(1)
-      else:
+      # TODO-64B:
+      # Assumes the product of the two sizes is <4GB here.
```

Comment:
Can you add an assert to verify the product is small enough? Will help catch problems later in case we forget.

Reply:
we need one on the host side - don't want to add logic in the kernel if we can avoid it.